File: backend/models/resume.py
from datetime import datetime
from bson import ObjectId

class Resume:
    @staticmethod
    def create(user_id, job_role, summary='', skills=None, projects=None, experience=None, education=None):
        """Create a new resume document"""
        return {
            'user_id': ObjectId(user_id),
            'job_role': job_role,
            'summary': summary,
            'skills': skills or [],
            'projects': projects or [],
            'experience': experience or [],
            'education': education or [],
            'created_at': datetime.utcnow(),
            'updated_at': datetime.utcnow()
        }
    
    @staticmethod
    def update(updates):
        """Update resume document"""
        updates['updated_at'] = datetime.utcnow()
        return updates

File: src/types/resume.ts
export interface Project {
    title: string;
    description: string;
    technologies: string[];
    url?: string;
}

export interface Experience {
    company: string;
    position: string;
    duration: string;
    responsibilities: string[];
}

export interface Education {
    institution: string;
    degree: string;
    year: string;
}

export interface Resume {
    id: string;
    user_id: string;
    job_role: string;
    summary: string;
    skills: string[];
    projects: Project[];
    experience: Experience[];
    education: Education[];
    created_at: string;
    updated_at: string;
}

export interface CreateResumeData {
    job_role: string;
    summary?: string;
    skills?: string[];
    projects?: Project[];
    experience?: Experience[];
    education?: Education[];
}

export interface UpdateResumeData extends Partial<CreateResumeData> { }
